polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1170,12 +1165,9 @@ ValueCategory MLIRScanner::VisitAtomicExpr(clang::AtomicExpr *BO) {
 
 ValueCategory MLIRScanner::VisitExprWithCleanups(ExprWithCleanups *E) {
   auto ret = Visit(E->getSubExpr());
-  for (auto &child : E->children()) {
-    if (!SuppressWarnings) {
-      child->dump();
-      llvm::WithColor::warning() << "cleanup not handled\n";
-    }
-  }
+  for (auto &child : E->children())
+    mlirclang::warning() << "cleanup not handled for: " << child << "\n";
```

Comment:
`child->dump()` and `<< child` may actually print different results, can we do something like 
`child->dump(mlirclang::warning())`?